sotodlib/site_pipeline/make_focal_plane.py

```diff
@@ -0,0 +1,264 @@
+import argparse as ap
+import numpy as np
+import sotodlib.io.g3tsmurf_utils as g3u
+from sotodlib.core import AxisManager
+from scipy.spatial.transform import Rotation as R
+from pycpd import AffineRegistration
+
+
+def LAT_coord_transform(xy, rot_fp, rot_ufm, r=72.645):
+    """
+    Transform from instrument model coords to LAT Zemax coords
+
+    Arguments:
+
+        xy: XY coords from instrument model.
+            Should be a (2, n) array.
+
+        rot_fp: Angle of array location on focal plane in deg.
+
+        rot_ufm: Rotatation of UFM about its center.
+
+    Returns:
+
+        xy_trans: Transformed coords.
+    """
+    xy_trans = np.zeros((xy.shape[1], 3))
+    xy_trans[:, :2] = xy.T
+
+    r1 = R.from_euler("z", rot_fp, degrees=True)
+    shift = r1.apply(np.array([r, 0, 0]))
+
+    r2 = R.from_euler("z", rot_ufm, degrees=True)
+    xy_trans = r2.apply(xy_trans) + shift
+
+    return xy_trans.T[:2]
+
+
+def rescale(xy):
+    """
+    Rescale pointing or template to [0, 1]
+
+    Arguments:
+
+        xy: Pointing or template, should have two columns.
+
+    Returns:
+
+        xy_rs: Rescaled array.
+    """
+    xy_rs = xy.copy()
+    xy_rs[:, 0] /= xy[:, 0].max() - xy[:, 0].min()
+    xy_rs[:, 0] -= xy_rs[:, 0].min()
+    xy_rs[:, 1] /= xy[:, 1].max() - xy[:, 1].min()
+    xy_rs[:, 1] -= xy_rs[:, 1].min()
+    return xy_rs
+
+
+def gen_priors(aman, prior, method="flat", width=1, basis=None):
+    """
+    Generate priors from detmap.
+    Currently priors will be 1 everywhere but the location of the detmap's match.
```

Comment:
I guess the location of the detmap's match is infinite? Meaning that we trust the detmap result 100% for now?

Reply:
It's whatever gets passed in as `prior` right now. In the case where we want to trust the detmap 100% there is a flag that can be passed that will skip all the fitting stuff and just copy the `det_id` from the detmap.